extensions/platform-http/component/src/main/java/org/apache/camel/component/http/server/PlatformHttpComponent.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.http.server;
+
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.Endpoint;
+import org.apache.camel.component.http.server.spi.Method;
+import org.apache.camel.support.DefaultComponent;
+
+public class PlatformHttpComponent extends DefaultComponent {
+
+    public PlatformHttpComponent() {
+        super();
+    }
+
+    public PlatformHttpComponent(CamelContext context) {
+        super(context);
+    }
+
+    @Override
+    protected Endpoint createEndpoint(String uri, String remaining, Map<String, Object> parameters) throws Exception {
+        final String[] remainingParts = remaining.split(":");
+        if (remainingParts.length != 2) {
+            throw new IllegalArgumentException("Expected two segments delimited by ':'; found " + remaining);
+        }
+        final Set<Method> methods = Method.parseList(remainingParts[0]);
+        final String path = remainingParts[1];
+        return new PlatformHttpEndpoint(uri, this, methods, path);
```

Comment:
I would prefer the URI syntax to be similar to other components, like 

    platform-http://relative_path[?options]

Also follow the conventions other http components have to filter methods (see `httpMethodRestrict`)

Reply:
You propose more typing but less learning for those who know the older Camel components.

Reply:
It is about consistency

Reply:
Let's allow methods before the path and also as an uriParam to have both good ergonomy and compatibilty with the older components. So all of the following variants would yeld the same result: 

```
platform-http:/path?httpMethodRestrict=GET,POST
platform-http:GET:/path?httpMethodRestrict=POST
platform-http:GET,POST:/path
```